packages/google-auth-library-nodejs/test/gtoken/test.getCredentials.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {describe, it, after} from 'mocha';
+import * as fs from 'fs';
+import {getCredentials} from '../../src/gtoken/getCredentials';
+
+describe('getCredentials', () => {
+  const filePaths: string[] = [];
+
+  function writeTempFile(name: string, data: string): string {
+    const filePath = `./${name}`;
+    fs.writeFileSync(filePath, data);
+    filePaths.push(filePath);
+    return filePath;
+  }
```

Comment:
Maybe we could use [os.tmpdir()](https://nodejs.org/api/os.html\#ostmpdir) instead? I generally avoid creating temporary files during tests to prevent potential spillage. While this is less critical when tests run in an ephemeral environment, if files do spill and someone runs the tests locally, they will end up with undesired directories.